subscription/src/main/java/org/killbill/billing/subscription/api/user/DefaultSubscriptionBaseApiService.java

```diff
@@ -669,6 +670,51 @@ public int handleBasePlanEvent(final DefaultSubscriptionBase subscription, final
         }
     }
 
+    @Override
+    public int handleExpiredEvent(final DefaultSubscriptionBase subscription, final SubscriptionBaseEvent event, final SubscriptionCatalog catalog, final CallContext context) throws CatalogApiException {
+
+        final InternalCallContext internalCallContext = createCallContextFromBundleId(subscription.getBundleId(), context);
+        if (event.getType() == EventType.EXPIRED && subscription.getCategory() == ProductCategory.BASE) {
+
+            final List<SubscriptionBaseEvent> expireEvents = new LinkedList<SubscriptionBaseEvent>();
+            final List<DefaultSubscriptionBase> subscriptionsToBeExpired = computeAddOnsToExpire(expireEvents, subscription.getBundleId(), event.getEffectiveDate(), catalog, internalCallContext);
+            dao.cancelSubscriptionsOnBasePlanEvent(subscription, event, subscriptionsToBeExpired, expireEvents, catalog, internalCallContext); //TODO_1533, not renamed this method to cancelOrExpireSubscriptionsOnBasePlanEvent as there are many other methods in the DAO.
```

Comment:
Agreed for the renaming `cancelSubscriptionsOnBasePlanEvent` -> `cancelSubscriptionsOnNotificationEvent` (we should talk more about notification (event) than event, as event often implies bus event; our terminology here was confusing prior your change :-))

Reply:
okay, makes sense. How about renaming to `cancelOrExpireSubscriptionOnNotification` ?